agent/config/config_test.go

```diff
@@ -592,6 +601,49 @@ func TestUserDataConfig(t *testing.T) {
 	}
 }
 
+func TestContainerInstancePropagateTags(t *testing.T) {
+	testcases := []struct {
+		name                                   string
+		envVarVal                              string
+		expectedContainerInstancePropagateTags ContainerInstancePropagateTagsType
+	}{
+		{
+			name:      "none container instance propagate tags",
+			envVarVal: "none",
+			expectedContainerInstancePropagateTags: ContainerInstancePropagateTagsNoneType,
+		},
+		{
+			name:      "ec2_instance container instance propagate tags",
+			envVarVal: "ec2_instance",
+			expectedContainerInstancePropagateTags: ContainerInstancePropagateTagsEC2InstanceType,
+		},
+		{
+			name:      "invalid container instance propagate tags",
+			envVarVal: "none",
+			expectedContainerInstancePropagateTags: ContainerInstancePropagateTagsNoneType,
+		},
+	}
+
+	for _, tc := range testcases {
+		// Test the parse function only.
+		t.Run(tc.name, func(t *testing.T) {
```

Comment:
I'd probably merge these two t.Run functions together and just make sure the "message" includes a note that the parsing didn't work as expected.